src/Microsoft.Health.Fhir.Liquid.Converter/Hl7v2/Hl7v2TemplateProvider.cs

```diff
@@ -16,52 +15,34 @@ namespace Microsoft.Health.Fhir.Liquid.Converter.Hl7v2
 {
     public class Hl7v2TemplateProvider : MemoryFileSystem, ITemplateProvider
     {
-        private const string TemplateFileExtension = ".liquid";
-
         public Hl7v2TemplateProvider(string templateDirectory)
         {
-            TemplateCollection = LoadTemplates(templateDirectory);
+            if (!Directory.Exists(templateDirectory))
+            {
+                throw new ConverterInitializeException(FhirConverterErrorCode.TemplateFolderNotFound, string.Format(Resources.TemplateFolderNotFound, templateDirectory));
+            }
+
+            TemplateDirectory = templateDirectory;
+            TemplateCollection = LoadCodeSystemMapping();
         }
 
         public Hl7v2TemplateProvider(List<Dictionary<string, Template>> templateCollection)
```

Comment:
How about deep-copying the `templateCollection` before it being assigned to `TemplateCollection`? Or the invoker might be able to manage the internal state under directly assigning mode.

Reply:
Thanks, fixed in the update.

Reply:
By offline discussion, we plan to copy list of dictionaries now.